accounts-db/src/append_vec.rs

```diff
@@ -1668,26 +1637,31 @@ pub mod tests {
             // wrap AppendVec in ManuallyDrop to ensure we do not remove the backing file when dropped
             let av = ManuallyDrop::new(AppendVec::new(path, true, 1024 * 1024));
 
-            let crafted_data_len = 1;
-
             av.append_account_test(&create_test_account(10)).unwrap();
-
-            av.get_stored_account_meta_callback(0, |account| {
-                let StoredAccountMeta::AppendVec(account) = account else {
-                    panic!("StoredAccountMeta can only be AppendVec in this test.");
-                };
-                account.set_data_len_unsafe(crafted_data_len);
-                assert_eq!(account.data_len(), crafted_data_len);
-
-                // Reload accounts and observe crafted_data_len
-                av.get_stored_account_meta_callback(0, |account| {
-                    assert_eq!(account.data_len() as u64, crafted_data_len);
-                });
-            });
-
             av.flush().unwrap();
             av.len()
         };
+
+        // Assert that the file is currently valid.
+        {
+            let av =
+                ManuallyDrop::new(AppendVec::new_from_file(path, accounts_len, storage_access));
+            assert!(av.is_ok());
+        }
+
+        // Manually manipulate the `data_len` bytes of the first account.
+        {
+            const ACCOUNT_0_DATA_LEN_OFFSET: u64 =
```

Comment:
Can this const be computed once for all tests?

Reply:
ec1c69a42a